src/webhook_launcher/app/models.py

```diff
@@ -190,6 +190,18 @@ def matches(self, proj_name):
                 return True
         return False
 
+    @classmethod
+    def get_matching(cls, name, apiurl):
+        # Search all Projects for a match
+        # Exception raised if more than one match found
+        found=None
+        for project in Project.objects.filter(obs__apiurl=apiurl):
+            if project.matches(name):
+                if found is not None:
+                    raise MultipleObjectsReturned("Project %s matches both %s and %s (possibly more)" %
+                                                  (name, found.name, project.name))
+                found = project
+        return found
```

Comment:
Project class has a 'match' attribute which is documented (and used) to "If set then used as well as name to re.match() project names"
Which is awesome and all that when you have a Project and want to decide if a name belongs to it.... not so awesome if you have a name and want to find a Project that claims it to see if it is official or gated or ... (see code in WebhookMapping.project_disabled() in this file)
This implementation will clearly not scale beyond a few Projects